Gok 211-This PR handles the Save functionality in stockReceipt tab.

deepthi-mantena opened the pull request.

deepthi-mantena committed:
[Deepthi M/Abinaya] |GOK-187 | Updated to fetch list of patients with activedrug orders to display in dispense tab.

deepthi-mantena committed:
[Deepthi M/Abinaya] |GOK-187 | Updated to fix Lint issue.

deepthi-mantena committed:
[Deepthi M/Abinaya] |GOK-187 | Updated to add patientID.

abinaya-u committed:
GOK-28 | add. dispense functionality
Co-authored-by: deepthi-m <deepthi.mantena@thoughtworks.com>

abinaya-u committed:
add. stock receipt functionality
Co-authored-by: deepthi-m <deepthi.mantena@thoughtworks.com>

abinaya-u committed:
update. stockReceipt url and functioanlity
Co-authored-by: deepthi-m <deepthi.mantena@thoughtworks.com>

deepthi-mantena committed:
[Deepthi M/Abinaya] |stock-receipt | Added save and cancel buttons

deepthi-mantena committed:
Merge branch 'master' into stock-receipt

abinaya-u committed:
GOK-89 | WIP | add. functionality to display stock receipt screen

Co-authored-by: deepthi-m <deepthi.mantena@thoughtworks.com>

abinaya-u committed:
GOK-89 | add. cancel functionality for stock-receipt
Co-authored-by: deepthi-m <deepthi.mantena@thoughtworks.com>

abinaya-u committed:
GOK-89 | update. styles and responsiveness for stock-receipt
Co-authored-by: deepthi-m <deepthi.mantena@thoughtworks.com>

abinaya-u committed:
GOK-89 | refactor. made item column sticky and stock receipt page responsive

abinaya-u committed:
GOK-89 | refactor. made item column sticky and stock receipt page responsive

abinaya-u committed:
GOK-89 | fix. lint issue

abinaya-u committed:
GOK-89 | refactor. move bussiness logic to separate file

Co-authored-by: deepthi-m <deepthi.mantena@thoughtworks.com>

abinaya-u committed:
fix. lint issues
Co-authored-by: deepthi-m <deepthi.mantena@thoughtworks.com>

deepthi-mantena committed:
Deepthi M| GOK-211 |Added the save functionality for the stock receipt tab.

deepthi-mantena committed:
Deepthi M| GOK-211 |Updated to fix build issue

deepthi-mantena committed:
Revert "Deepthi M| GOK-211 |Updated to fix build issue"

This reverts commit 7ca4705bbd495d5c37606ff6f6778b93edad819d.

deepthi-mantena committed:
Revert "Revert "Deepthi M| GOK-211 |Updated to fix build issue""

This reverts commit 91580f4ec5eabddff7abc9408b69d3a9248419b7.

deepthi-mantena committed:
Deepthi M| GOK-211 |Updated to fix build issue

deepthi-mantena committed:
Deepthi M| GOK-211 |Updated to fix build issue

gsluthra commented:
Please provide some description on what all tasks this code does and if possible, also provide a screenshot. thanks!

deepthi-mantena commented:
@gsluthra As part of this PR I have modified save-receipt.js stock-receipt.jsx and apiUtils.js to handle save by constructing the request body by making respective api calls and populating the values of Requestbody attributes.The below screen shot shows the basic landing page which would display on click of stockfetch button after entering an outwardID
<img width="1672" alt="Screenshot 2023-07-24 at 9 46 01 PM" src="https://github.com/Bahmni-HWC/bahmnihwc-inventory-frontend/assets/96411257/a5aa254f-7ef1-4291-9323-0cb88afc745c">
Now I updated Batchquantity to 5 
<img width="1631" alt="Screenshot 2023-07-24 at 9 51 21 PM" src="https://github.com/Bahmni-HWC/bahmnihwc-inventory-frontend/assets/96411257/d240c0d4-4bb2-4fb3-b415-900b22a93d78">
Now in Db these values shown On Ui should be saved in appropriate columns.

<img width="1073" alt="Screenshot 2023-07-24 at 9 52 58 PM" src="https://github.com/Bahmni-HWC/bahmnihwc-inventory-frontend/assets/96411257/c67cf4fc-ba25-43cc-b6d6-3aa3e8df5a72">
<img width="1222" alt="Screenshot 2023-07-24 at 9 54 29 PM" src="https://github.com/Bahmni-HWC/bahmnihwc-inventory-frontend/assets/96411257/39db4f75-1fc4-4726-9ebe-56e0453289e5">

deepthi-mantena committed:
Merge branch 'master' into GOK-211

deepthi-mantena committed:
Deepthi M| GOK-211 |Updated to resolve merge conflicts

deepthi-mantena committed:
Deepthi M| GOK-211 |Removed helper.js file

gsluthra commented on src/service/save-receipt.js:
```diff
@@ -1,29 +1,46 @@
+       const instanceTypeResponse = await getRequest(`/openmrs/ws/rest/v2/inventory/stockOperationType?v=full&q=Receipt`);
+       const instanceTypeUuids = instanceTypeResponse.results[0].uuid;
+       const currentDate = new Date();
+       const formattedDate = `${currentDate.getDate().toString().padStart(2, '0')}-${
```
Ideally should be in a date-utils.js. And should have a unit test.

deepthi-mantena replied:
@gsluthra as suggested moved date formatting to dateutils.js and updated PR .For unittest may be we will create a tech debt card.

deepthi-mantena committed:
Deepthi M| GOK-211 |Moved the date formatting to a util class

deepthi-mantena committed:
Deepthi M| GOK-211 |Updated to fix Lint issue

deepthi-mantena committed:
Deepthi M| GOK-211 |updated to fix issue with formatted date

gsluthra commented on src/utils/date-utils.js:
```diff
@@ -0,0 +1,9 @@
+const getFormattedDate = () => {
```
Add a comment which explains what does this method do. thanks!

deepthi-mantena replied:
Yes @gsluthra added the comments.

deepthi-mantena committed:
Deepthi M| GOK-211 |updated comments for the formatted date function
